opensearch-operator/pkg/builders/cluster.go

```diff
@@ -446,6 +446,15 @@ func NewSTSForNodePool(
 		initContainers = append(initContainers, keystoreInitContainer)
 	}
 
+	discoveryType := ""
+	initialMasterNodes := BootstrapPodName(cr)
+
+	// Support for single-node clusters
+	if cr.Spec.NodePools[0].Replicas < 2 {
+		discoveryType = "single-node"
+		initialMasterNodes = ""
+	}
```

Comment:
What about when it's 2 replicas?
By what I'm reading it's going to work only with =1 or >=3 replicas, =2 would break things...